### PR TITLE
WIndowsPb: gpg signature verification for ANT binary

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [x] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/2908

It seems only the Ant download supplies a .asc or .sig file. For the other downloads, we supply a fixed checksum string which, since we do not plan on changing, provide adequate verification of a download. 

Theres still the java downloads in the playbook which are capable of having gpg verification. Getting the signature file is a bit tricky, on unix we do something like this 

```
curl -s 'https://api.adoptium.net/v3/assets/feature_releases/{{ jdk_version }}/ga?architecture={{ api_architecture }}&heap_size=normal&image_type=jdk&jvm_impl={{ bootjdk }}&os={{ platformLinux }}&page=0&page_size=1&project=jdk&vendor=eclipse' | grep signature_link | awk '{split($0,a,"\""); print a[4]}'
```
which does not seem to be working well in a windows environment. So until I can figure this out, I thought id merge my work so far

### CODE REVIEW

**Comment by Haroon-Khel** [comment]
windows vpc https://ci.adoptium.net/job/VagrantPlaybookCheck/1661/console

**Comment by Haroon-Khel** [comment]
Latest vpc https://ci.adoptium.net/job/VagrantPlaybookCheck/1662/console

**Comment by sxa** [comment]
You can set an environment variable `GNUPGHOME=$WORKSPACE/.gnupg` to a specific directory to use, so that may give you a way forward to use an alternate location (I use this for the Temurin GPG signing to store it in the jenkins workspace)

**Comment by Haroon-Khel** [comment]
> I would guess that's because it's running under cmd.exe (The windows shell) and not cygwin's bash and therefore doesn't understand /home/vagrant. When you tested outside ansible was it from a cygwin shell?

Works in a cmd.exe environment too

image

**Comment by Haroon-Khel** [comment]
> (There's probably a command line option to set it too, but I've not used that)

--homedir looks like it should work, just need to figure out how to use it properly

image

(both forward and backslash path locations)

**Comment by sxa** [comment]
Interesting ... the other thing to check would be whether there is another `gpg` tool on the machine that's being picked up instead of the cygwin one. Could run a test where you fully qualify `c:\cygwin\bin\gpg` to run it and see if that makes a difference.

**Review by sxa — APPROVED** [review]
LGTM but my only comment would be that it may not be necessary to import the key each time (e.g. it could be skipped if it already exists on the machine)

**Comment by karianna** [comment]
Playbook checkers are failing, related?

**Comment by sxa** [comment]
> Playbook checkers are failing, related?

Looks like it. (My review was based on reading the code, and so the check failures need to be understood and addressed before this is merged)

**Comment by karianna** [comment]
@Haroon-Khel needs a linter fix

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/ANT/tasks/main.yml [hunk 1]
```diff
@@ -26,6 +26,16 @@
   register: ant_download
   tags: ANT
 
+- name: GPG Signature Verification of Binary
+  import_tasks:
+    file: "{{playbook_dir}}/roles/GPG_signature_verification/tasks/main.yml"
+  vars:
+    file_path: c:\temp\ant.zip
+    signature_link: "https://archive.apache.org/dist/ant/binaries/apache-ant-1.10.5-bin.zip.asc"
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/ANT/tasks/main.yml:34]: Ant is on 1.10.13 now - should we upgrade?

**Review by karianna — COMMENTED** [review]
Ant version

**Comment by Haroon-Khel** [comment]
Ant contrib is failing to download on the 2022 windows workflow
```
ASK [ANT : Download ant-contrib] **********************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "dest": "c:\\temp\\ant-contrib.zip", "elapsed": 0.8295483, "msg": "Error downloading 'https://sourceforge.net/projects/ant-contrib/files/ant-contrib/ant-contrib-1.0b2/ant-contrib-1.0b2-bin.zip' to 'c:\\temp\\ant-contrib.zip': The request was aborted: Could not create SSL/TLS secure channel.", "status_code": null, "url": "https://sourceforge.net/projects/ant-contrib/files/ant-contrib/ant-contrib-1.0b2/ant-contrib-1.0b2-bin.zip"}
```
however that error is unrelated to this pr, the gpg check on the ant binary just before it passes without error

A bit confused by this linter error
```

Run # Ansible code static analysis
WARNING  Listing 1 violation(s) that are fatal
syntax-check[missing-file]: Unable to retrieve file contents
ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/ANT:1:1 Could not find or access '/tmp/roles/GPG_signature_verification/tasks/main.yml' on the Ansible Controller.

Error: Unable to retrieve file contents

                    Rule Violation Summary                     
 count tag                        profile rule associated tags 
     1 syntax-check[missing-file] min     core, unskippable    

Failed after : 1 failure(s), 0 warning(s) on 505 files.
Error: Process completed with exit code 2.
```

**Comment by Haroon-Khel** [comment]
Linter issue is solved. The pr checks are running now because I had to rebase (branch was out of date) but this pr is ready to be merged

**Comment by Haroon-Khel** [comment]
The ant contrib error mentioned in https://github.com/adoptium/infrastructure/pull/3018#issuecomment-1558971620 is intermittent

**Pull request merged**